.github/workflows/matter-validation.yml

```diff
@@ -0,0 +1,68 @@
+name: Matter Validation
+
+on:
+  pull_request:
+    branches:
+      - main
+      - 'v*-branch'
+
+    paths:
+      - 'samples/matter/**'
+      - 'applications/matter_bridge/**'
+      - 'applications/matter_weather_station/**'
+
+permissions:
+  contents: read
+
+jobs:
+  matter-validation:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v5
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+          path: ncs/nrf
+
+      - name: Rebase
```

Comment:
actions/checkout step can clone merged version of repository. Is rebase needed for some reason, or can this be simplified?